### PR TITLE
docs(compiler): fix typos and spelling

### PR DESCRIPTION
**Pull request by arturovt**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/compiler/design/separate_compilation.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ AUTHOR: chuckj@
 
 ### Angular 5 (Renderer2)
 
-In 5.0 and prior versions of Angular the compiler performs whole program
+In 5.0 and prior versions of Angular the compiler performs the whole program
```
Comment (petebacondarwin) [inline packages/compiler/design/separate_compilation.md]: This change is not correct. Please revert it.

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
One small fix but the rest LGTM

File: packages/compiler/design/separate_compilation.md (hunk 1 above)
Comment (arturovt) [reply packages/compiler/design/separate_compilation.md]: @petebacondarwin Done.

**Review by petebacondarwin — APPROVED** [review]

**Comment by kapunahelewong** [comment]
Thank you for these fixes, @arturovt! 🌺

**Comment by kapunahelewong** [comment]
merge-assistance: Global approval for this docs-only change.

**Pull request closed without merging**